play_motion:
  motions:
    cleanup_look_down:
      joints: [torso_lift_joint, head_1_joint, head_2_joint]
      points:
      - positions: [0.3, 0.0, -0.4]
        time_from_start: 2.0
      
    back_to_default:
      joints: [torso_lift_joint, head_1_joint, head_2_joint]
      points:
      - positions: [0.22, 0.0, 0.0]
        time_from_start: 2.0

    low_confirmation_mode:
      joints: [torso_lift_joint, head_1_joint, head_2_joint]
      points:
      - positions: [0.0, 0.0, -0.8]
        time_from_start: 2.0

    low_pickup_mode:
      joints: [torso_lift_joint, head_1_joint, head_2_joint, arm_1_joint, arm_2_joint, arm_3_joint, arm_4_joint, arm_5_joint, arm_6_joint, arm_7_joint]
      points:
      - positions: [0.0, 0.0, -0.8, 0.21, 0.35, -0.2, 0.8, -1.57, 1.37, 0.0]
        time_from_start: 2.0

    medium_mode:
      joints: [torso_lift_joint, head_1_joint, head_2_joint]
      points:
      - positions: [0.08, 0.0, -0.6]
        time_from_start: 2.0

    high_mode:
      joints: [torso_lift_joint, head_1_joint, head_2_joint]
      points:
      - positions: [0.35, 0.0, 0.0]
        time_from_start: 2.0

    head_look_around:
      joints: [head_1_joint, head_2_joint]
      points:
      - positions: [0, -0.7]
        time_from_start: 0.1
      - positions: [-0.7, -0.7]
        time_from_start: 1.0
      - positions: [-0.7, -0.5]
        time_from_start: 2.0
      - positions: [0, -0.5]
        time_from_start: 3.0
      - positions: [0.7, -0.5]
        time_from_start: 4.0
      - positions: [0.7, -0.7]
        time_from_start: 5.0
      - positions: [0, -0.6]
        time_from_start: 6.0
    
    holding_object:
      joints: [torso_lift_joint, head_1_joint, head_2_joint, arm_1_joint, arm_2_joint, arm_3_joint, arm_4_joint, arm_5_joint, arm_6_joint, arm_7_joint]
      points:
      - positions: [0.07, 0.0, 0.0, 0.08, -1.24, -0.08, 2.16, 0.08, 1.48, -1.75]
        time_from_start: 2.0

    look_bin:
      joints: [head_1_joint, head_2_joint]
      points:
      - positions: [0.0, -0.85]
        time_from_start: 2.0

    open_gripper:
      joints: [gripper_left_finger_joint, gripper_right_finger_joint]
      points:
      - positions: [0.04, 0.04]
      time_from_start: 2.0

# Defining joint positions for placing an item on tiagos back
# Values used to experiement with stuff, nothing is really used
TIAGo_back_RHS: { arm_1_joint: 0.08,
                  arm_2_joint: 0.16,
                  arm_3_joint: -3.46,
                  arm_4_joint: 1.85,
                  arm_5_joint: 1.9,
                  arm_6_joint: -1.02, 
                  arm_7_joint: 0.16 }

TIAGo_back_LHS: { arm_1_joint: 2.45,
                  arm_2_joint: 0.88,
                  arm_3_joint: -2.51,
                  arm_4_joint: 2.00,
                  arm_5_joint: -0.49,
                  arm_6_joint: 0.87, 
                  arm_7_joint: -1.31 }

TIAGo_back_LHS2: { arm_1_joint: 2.61,
                  arm_2_joint: 0.66,
                  arm_3_joint: -2.57,
                  arm_4_joint: 1.8,
                  arm_5_joint: -0.96,
                  arm_6_joint: 0.91, 
                  arm_7_joint: 0.75 }

TIAGo_base_LHS: { arm_1_joint: 1.55,
                  arm_2_joint: -0.92,
                  arm_3_joint: -0.76,
                  arm_4_joint: 1.82,
                  arm_5_joint: 0.61,
                  arm_6_joint: -0.79, 
                  arm_7_joint: 1.36 }     

RHS_position: { x: 0.019, y: -0.259, z: 1.351 }
RHS_orientation: { x: 0.3766, y: 0.603, z: 0.6355, w: 0.3003 }

LHS_position: { x: -0.4108, y: 0.1437, z: 1.25 }
LHS_orientation: { x: -0.378998, y: 0.56025, z: -0.61733, w: 0.4017 }

LHS2_position: { x: -0.0088, y: 0.2087, z: 1.25 }
LHS2_orientation: { x: -0.381976, y: 0.5585, z: -0.6163, w: 0.40287 }


# - positions: [0.04, -0.5, 0.21, -0.2, -2.2, 1.15, -1.57, 0.2, 0.0]
      #   time_from_start: 6.0 for low mode